File: c++/functions/returning_value.cpp
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */
// SPDX-License-Identifier: GPL-3.0-or-later

#include <iostream>

bool returning_boolean()
{
  return true;
}

int returning_integer()
{
  return 7;
}

const char* returning_string()
{
  return "hello";
}

int main()
{
  bool b;
  int i;
  const char* s;

  b = returning_boolean();
  std::cout << "boolean:\n"
    << "b = " << b
    << std::endl << std::endl;

  i = returning_integer();
  std::cout << "integer:\n"
    << "i = " << i
    << std::endl << std::endl;

  s = returning_string();
  std::cout << "string:\n"
    << "s = " << s
    << std::endl << std::endl;
}

File: javascript/functions/returning_value.js
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

function returning_boolean() {
  return true;
}

function returning_integer() {
  return 7;
}

function returning_string() {
  return "hello";
}

let b = returning_boolean();
console.log(`boolean:\n`
  + `b = ${b}\n`);

let i = returning_integer();
console.log(`integer:\n`
  + `i = ${i}\n`);

let s = returning_string();
console.log(`string:\n`
  + `s = ${s}\n`);
